core/src/com/mygdx/game/pathfinding/Triangle.java

```diff
@@ -103,11 +105,16 @@ public Vector3 getRandomPoint(Vector3 out) {
 	 * @return
 	 */
 	public float area() {
-		double ab = a.dst(b);
-		double bc = b.dst(c);
-		double ca = a.dst(c);
-		double s = (ab + bc + ca) / 2;
-		return (float) Math.sqrt(s * (s - ab) * (s - bc) * (s - ca));
+		final float abx = b.x - a.x;
+		final float aby = b.y - a.y;
+		final float abz = b.z - a.z;
+		final float acx = c.x - a.x;
+		final float acy = c.y - a.y;
+		final float acz = c.z - a.z;
+		final float r = aby * acz - abz * acy;
+		final float s = abz * acx - abx * acz;
+		final float t = abx * acy - aby * acx;
+		return 0.5f * (float) Math.sqrt(r * r + s * s + t * t);
```

Comment:
One sqrt instead of 4 (each dst hides a sqrt).
Also the original code returns NaN in certain situations (I suspect that the argument of the last sqrt can become negative due to rounding errors). Looks like this problem doesn't happen with the new code.